agent/agent.go

```diff
@@ -123,24 +119,8 @@ func (a *Agent) handleRequest(ctx *spcontext.Context, id string, msg *privatevcs
 		}
 	}
 
-	ctx = ctx.With(
-		"id", id,
-		"pool_id", a.poolConfig.PoolULID,
-		"method", req.Method,
-		"raw_path", req.URL.EscapedPath(),
-		"path", req.URL.Path,
-	)
-
-	for key, value := range msg.Headers {
-		req.Header.Set(key, value)
-	}
-
-	name, project, subdomain, err := validation.MatchRequest(a.vendor, req)
+	ctx, err = a.validator.Validate(ctx, a.vendor, req)
```

Comment:
I think we might want to still add `method`, `path`, etc to the context in here rather than relying on the validator to do so. The reason is that we log successful as well as failed requests, and want to make sure we're logging all the useful info.

Reply:
Correct, I removed a bit too much.